middleware/moonraker_client.py

```
@@ -0,0 +1,178 @@
+
+
+def list_objects(moonraker: str, timeout: float = DEFAULT_TIMEOUT) -> list | None:
+    """GET /printer/objects/list. Returns the object name list, or None on failure."""
+    if not moonraker:
+        return None
+    try:
+        response = requests.get(f"{moonraker}/printer/objects/list", timeout=timeout)
+        response.raise_for_status()
+        objects = response.json().get("result", {}).get("objects", [])
+        return objects if isinstance(objects, list) else None
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)